### ISSUE TITLE
httpNobody incorrect suggest/fix

### ISSUE THREAD

**Issue by pierrre** [go-critic/go-critic#1256]
I'm using go-critic with golangci-lint v1.49.0.
It uses go-critic v0.6.4.

This code
```go
http.NewRequestWithContext(ctx, http.MethodGet, "http://localhost", nil)
```
is flagged by the `httpNobody` rule.
That's the expected behavior.

However the "--fix" golangci-lint option incorrectly rewrite it to:
```go
http.NewRequestWithContext(ctx, http.MethodGet, "http://localhost", http.NewRequestWithContext(ctx, http.MethodGet, "http://localhost", http.NoBody))
```

I guess it uses the suggestion from
https://github.com/go-critic/go-critic/blob/5c89720acc9e12fbbc23d95b381ee3b594762088/checkers/rules/rules.go#L124-L128

But I don't see any obvious issue.

Is there a way to check what is the suggestion output when I run go-critic manually ?

**Comment by hansgylling** [comment]
Is this a duplicate of bug #1203 ?

**Comment by pierrre** [comment]
yes you're right
sorry, my bad
